Sheet 2/ProblemF.cpp

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <cmath>
+
+using namespace std;
+
+int main()
+{
+    int a, b, d = 0;
+
+    cin >> a >> b;
+
+    for (int i = 0; i <= 32; i++)
```

Comment:
Are you sure this shouldn't stop at `< 32`?

E.g., `pow(2, 32)` would exceed the capacity of `int`.

You're lucky here because `x` & `y` will be 0 after ~30 & thus will never enter the if-condition, but the principle stands.